File: lib/src/enum/download_status.dart
/// Download Status
/// This enum is used to define the status of the download task.
enum DownloadStatus {
  idle(name: "idle"),

  /// The download task is pending.
  pending(name: "pending"),

  /// The download task is downloading.
  downloading(name: "downloading"),

  /// The download task is completed.
  completed(name: "completed"),

  /// The download task is failed.
  failed(name: "failed"),

  /// The download task is canceled.
  canceled(name: "canceled"),

  paused(name: "paused");

  final String name;

  const DownloadStatus({required this.name});

  /// Get the download status from the name.
  static DownloadStatus fromName(String name) {
    return DownloadStatus.values.firstWhere(
      (e) => e.name.toLowerCase().trim() == name.toLowerCase().trim(),
      orElse: () => throw Exception('Download status not found: $name'),
    );
  }

  /// Get if the task should be cleared after completion.
  bool get clearTaskAfterCompletion =>
      this == completed || this == failed || this == canceled;
}

File: example/ios/Runner/Service/Models/DownloadStatus.swift
//
//  DownloadStatus 2.swift
//  Developer
//
//  Created by Nguyễn Minh Hưng on 5/5/26.
//


enum DownloadStatus {
    case idle
    case pending
    case inProgress
    case completed
    case failed
    case canceled
    case paused
}


extension DownloadStatus {
    var isInProgress: Bool {
        return self == .inProgress
    }
    
    var isCompleted: Bool {
        return self == .completed
    }
    
    var isPaused: Bool {
        return self == .paused
    }
}
